src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/ViewModels/ValidationProblemViewModelTestExtensions.cs

```diff
@@ -0,0 +1,244 @@
+        return error;
+    }
+
+    public static ErrorViewModel AssertHasSortMaxFieldLengthError(
+        this ValidationProblemViewModel validationProblem,
+        string expectedPath,
+        string field,
+        int maxFieldLength = 40)
+    {
+        var error = validationProblem.AssertHasError(
+            expectedPath: expectedPath,
+            expectedCode: ValidationMessages.SortMaxFieldLength.Code
+        );
+
+        var errorDetail = error.GetDetail<SortStringValidators.MaxFieldLengthErrorDetail>();
+
+        Assert.Equal(field, errorDetail.Value.Field);
+        Assert.Equal(maxFieldLength, errorDetail.MaxFieldLength);
+
+        return error;
+    }
+
+    public static ErrorViewModel AssertHasSortDirectionError(
+        this ValidationProblemViewModel validationProblem,
+        string expectedPath,
+        string direction)
+    {
+        var error = validationProblem.AssertHasError(
+            expectedPath: expectedPath,
+            expectedCode: ValidationMessages.SortDirection.Code
+        );
+
+        var errorDetail = error.GetDetail<SortStringValidators.DirectionErrorDetail>();
+
+        Assert.Equal(direction, errorDetail.Value.Direction);
+        Assert.Equal(["Asc", "Desc"], errorDetail.AllowedDirections);
+
+        return error;
+    }
+
+    public static ErrorViewModel AssertHasSortFieldsNotFoundError(
+        this ValidationProblemViewModel validationProblem,
+        string expectedPath,
+        IEnumerable<string> notFoundItems)
+    {
+        var error = validationProblem.AssertHasError(
+            expectedPath: expectedPath,
+            expectedCode: ValidationMessages.SortFieldsNotFound.Code
+        );
+
+        var errorDetail = error.GetDetail<NotFoundItemsErrorDetail<string>>();
+
+        Assert.Equal(notFoundItems, errorDetail.Items);
+
+        return error;
+    }
+
+    public static ErrorViewModel AssertHasSortFieldsNotFoundError(
+        this ValidationProblemViewModel validationProblem,
+        string expectedPath,
+        IEnumerable<DataSetQuerySort> notFoundItems)
+    {
+        var error = validationProblem.AssertHasError(
+            expectedPath: expectedPath,
+            expectedCode: ValidationMessages.SortFieldsNotFound.Code
+        );
+
+        var errorDetail = error.GetDetail<NotFoundItemsErrorDetail<DataSetQuerySort>>();
+
+        Assert.Equal(notFoundItems, errorDetail.Items);
+
+        return error;
+    }
+
+    public static ErrorViewModel AssertHasTimePeriodFormatError(
+        this ValidationProblemViewModel validationProblem,
+        string expectedPath,
+        string value)
+    {
+        var error = validationProblem.AssertHasError(
+            expectedPath: expectedPath,
+            expectedCode: ValidationMessages.TimePeriodFormat.Code
+        );
+
+        var errorDetail = error.GetDetail<FormatErrorDetail>();
+
+        Assert.Equal(value, errorDetail.Value);
+        Assert.Equal("{period}|{code}", errorDetail.ExpectedFormat);
```

Comment:
I'd potentially move this check into the tests in `TimePeriodStringValidatorsTests`. And then wouldn't bother checking it here. Then you have tests covering the fact that this `ExpectedFormat` is generated as expected, and this extension method is just concerned with the value.